docker-compose.yml

```diff
@@ -27,15 +27,4 @@ services:
     depends_on:
       - kafka
     links:
-      - kafka
-
-  php72:
```

Comment:
Before removing this we should decide on a solution how to test compatibility against the php with rdkafka ext. 

We might have multiple options here:

* Running the same integration tests also against the rdkafka extension (7.3 is now the latest php version supported by the extension). Downside: integration tests are not easy to maintain and will require deeper insight into the extension (which on the other hand I think  is good). The current tests need some more work to get them running....
* Running tests against interface specs. Downside: specs might be compatible but behavior might be different

Any other options?

Reply:
> Before removing this we should decide on a solution how to test compatibility against php with rdkafka ext.

Should compatibility to the rdkafka-ext really a requirement? Why should the world using an exact copy of an extension, when the difference is only the implementing language?

> integration tests are not easy to maintain

👍 

> specs might be compatible but behavior might be different

Could you explain what you mean? AFAIK specs [... describe the behaviour of an object you are about to write](https://www.phpspec.net/en/stable/manual/introduction.html)

Reply:
> Should compatibility to the rdkafka-ext really a requirement? Why should the world using an exact copy of an extension, when the difference is only the implementing language?

Good questions - i am thinking of that:

* a direct replacement does not lead to major changes in current integrations 
* composer dependency handling over extension upgrades - but still ffi extension is needed (which is not enabled by default)
* php native port of the c extension as a good start to learn more about librdkafka binding, ffi and the c background
* easier contribution (php vs c) - eg. to support more features faster like the admin stuff (sounds more like a counter argument... 😄)

> 
> > 7.3 is now the latest php version supported by the extension
> 
> Hmmmmm - **no**! It runs smoothly on 7.4

Hmmm. RdKafka ext uses nightly on travis - looks like master branch = 8.0.
https://github.com/arnaud-lb/php-rdkafka/blob/master/.travis.yml#L15

If it runs with 7.4 we might use the same image for both test cases (without / with rdkafka ext).

We might also think about turning back to php7.4 snapshot on travis when it is more reliable - maybe after the beta phase. I dropped it some while ago because it was not reliable and so i decided to use the local setup also within travis - simpler maintenance, but slower...

> Could you explain what you mean? AFAIK specs [... describe the behaviour of an object you are about to write](https://www.phpspec.net/en/stable/manual/introduction.html)

I just thought of basic interface tests - maybe something simply generic based on reflection and types... anything else might be more like integration tests then. I tend to still go for running the same tests also against the ext to see if it works the same....

Reply:
RdKafka ext builds successfully against 7.4snapshot now..